pkgs/applications/audio/zynaddsubfx/default.nix

```diff
@@ -1,30 +1,105 @@
-, libXpm, minixml, ntk, pkgconfig, zlib, liblo
+{ stdenv, lib, fetchFromGitHub, callPackage
+
+# Required build tools
+, cmake, pkgconfig
+
+# Required dependencies
+, fftw, liblo, minixml, zlib
+
+# Optional dependencies
+, alsaSupport ? true, alsaLib ? null
+, dssiSupport ? false, dssi ? null, ladspaH ? null
+, jackSupport ? true, libjack2 ? null
+, lashSupport ? true, lash ? null
+, ossSupport ? true
+, portaudioSupport ? true, portaudio ? null
+
+# Optional GUI dependencies
+, guiModule ? "off"
+, cairo ? null
+, fltk13 ? null
+, libGL ? null
+, libjpeg ? null
+, libX11 ? null
+, libXpm ? null
+, ntk ? null
+
+# Test dependencies
+, cxxtest
 }:
 
-stdenv.mkDerivation  rec {
-  pname = "zynaddsubfx";
+assert alsaSupport -> alsaLib != null;
+assert dssiSupport -> dssi != null && ladspaH != null;
+assert jackSupport -> libjack2 != null;
+assert lashSupport -> lash != null;
+assert portaudioSupport -> portaudio != null;
+
+assert builtins.any (g: guiModule == g) [ "fltk" "ntk" "zest" "off" ];
+assert guiModule == "fltk" -> fltk13 != null && libjpeg != null && libXpm != null;
+assert guiModule == "ntk" -> ntk != null && cairo != null && libXpm != null;
+assert guiModule == "zest" -> libGL != null && libX11 != null;
+
+stdenv.mkDerivation rec {
+  pname =
+    if guiModule == "zest" then "zyn-fusion"
+    else if guiModule != "off" then "zynaddsubfx-${guiModule}"
+    else "zynaddsubfx";
+
   version = "3.0.5";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/zynaddsubfx/zynaddsubfx-${version}.tar.bz2";
-    sha256 = "0qwzg14h043rmyf9jqdylxhyfy4sl0vsr0gjql51wjhid0i34ivl";
+  src = fetchFromGitHub {
+    owner = "zynaddsubfx";
+    repo = "zynaddsubfx";
+    rev = version;
+    sha256 = "1vh1gszgjxwn8m32rk5222z1j2cnjax0bqpag7b47v6i36p2q4x8";
+    fetchSubmodules = true;
   };
 
-  buildInputs = [ alsaLib cairo libjack2 fftw fltk13 lash libjpeg libXpm minixml ntk zlib liblo ];
-  nativeBuildInputs = [ cmake pkgconfig ];
-
   patchPhase = ''
     substituteInPlace src/Misc/Config.cpp --replace /usr $out
   '';
 
-  hardeningDisable = [ "format" ];
+  nativeBuildInputs = [ cmake pkgconfig ];
+  buildInputs = [ fftw liblo minixml zlib ]
+    ++ lib.optional alsaSupport alsaLib
+    ++ lib.optionals dssiSupport [ dssi ladspaH ]
+    ++ lib.optional jackSupport libjack2
+    ++ lib.optional lashSupport lash
+    ++ lib.optional portaudioSupport portaudio
+    ++ lib.optionals (guiModule == "fltk") [ fltk13 libjpeg libXpm ]
+    ++ lib.optionals (guiModule == "ntk") [ ntk cairo libXpm ]
+    ++ lib.optionals (guiModule == "zest") [ libGL libX11 ];
+
+  cmakeFlags = [ "-DGuiModule=${guiModule}" ]
+    # OSS library is included in glibc.
+    # Must explicitly disable if support is not wanted.
+    ++ lib.optional (!ossSupport) "-DOssEnable=OFF"
+
+    # Find FLTK without requiring an OpenGL library in buildInputs
+    ++ lib.optional (guiModule == "fltk") "-DFLTK_SKIP_OPENGL=ON"
```

Comment:
This wasn't necessary before, because `cairo` propagated `libglvnd`, but the fltk GUI doesn't use cairo.